### ISSUE TITLE
`LEFT OUTER JOIN` does not select opposite table's values (in MySQL does)

### ISSUE THREAD

**Issue by nkonev** [dolthub/dolt#5487]
Repro:

```
echo 'create database companies;' > /tmp/init.sql
docker run --rm -v /tmp/init.sql:/docker-entrypoint-initdb.d/init.sql -p 3306:3306 -t dolthub/dolt-sql-server:0.54.2
```

Than run the following SQL
```sql
create table legal_entity (
                              id binary(16) primary key,
    -- legal_entity
                              legal_entity_name               varchar(512) NOT NULL,
                              registration_number               varchar(255),
                              duns_number varchar(255),
                              vat_number varchar(255),
                              external_ids json not null,
                              traceability_state json,
                              attachments json not null
);


create table address(
                        id binary(16) primary key,
                        address_name varchar(512),
                        address_type varchar(32) not null,
                        address_line1        varchar(1024),
                        address_line2        varchar(1024),
                        city        varchar(255),
                        region        varchar(255),
                        code        varchar(255),
                        country json NOT NULL,
                        gln        varchar(255),
                        traceability_state json,
                        legal_entity_id binary(16),
                        foreign key (legal_entity_id) references legal_entity(id)
);

insert into legal_entity (id, legal_entity_name, registration_number, duns_number, vat_number, external_ids, traceability_state, attachments)
values  (uuid_to_bin('3634b1d8-8f1f-471d-b968-5c28d1c830f2'), 'Le1', 'wwwwwwwwww', '999999999', '12AAAAA1234AAZA', '{}', null, '[{"actor":"nikita.konev@example.com","createdAt":"2023-03-03T18:45:45","id":"26a16bf4-dd0e-479a-ba54-b9762cf3a9e5","name":"Booking Confirmation.pdf","url":"https://example.com/file/29e0232c-137b-496e-abee-13bdd1ad791e"}]');

insert into address (id, address_name, address_type, address_line1, address_line2, city, region, code, country, gln, traceability_state, legal_entity_id)
values  (uuid_to_bin('ce97477c-d1b1-41cc-9c29-ee5547e5fd50'), null, 'BUSINESS', '____1', null, 'Cy3', null, 'eee', '{"id":"BE","name":"Belgium"}', null, null, null),
        (uuid_to_bin('dba6c320-3625-47e1-a869-e8b6b1346588'), null, 'REGISTERED', 'w1', 'w2', 'cw', null, 'codw', '{"id":"CA","name":"Canada"}', null, null, uuid_to_bin('3634b1d8-8f1f-471d-b968-5c28d1c830f2'));


SELECT
    `legal_entity`.`vat_number` AS `vat_number`, `legal_entity`.`duns_number` AS `duns_number`, `legal_entity`.`id` AS `id`, `legal_entity`.`attachments` AS `attachments`, `legal_entity`.`external_ids` AS `external_ids`, `legal_entity`.`legal_entity_name` AS `legal_entity_name`, `legal_entity`.`traceability_state` AS `traceability_state`, `legal_entity`.`registration_number` AS `registration_number`, `address`.`gln` AS `address_gln`, `address`.`address_name` AS `address_address_name`, `address`.`address_type` AS `address_address_type`, `address`.`city` AS `address_city`, `address`.`code` AS `address_code`, `address`.`region` AS `address_region`,  `address`.`country` AS `address_country`, `address`.`id` AS `address_id`, `address`.`address_line2` AS `address_address_line2`, `address`.`address_line1` AS `address_address_line1`, `address`.`legal_entity_id` AS `address_legal_entity_id`, `address`.`traceability_state` AS `address_traceability_state`
FROM `legal_entity`
         LEFT OUTER JOIN `address` `address` ON `address`.`legal_entity_id` = `legal_entity`.`id`
WHERE `legal_entity`.`id` = uuid_to_bin('3634b1d8-8f1f-471d-b968-5c28d1c830f2');
```

It's seen - there is no `address`-related values in response:
![Screenshot from 2023-03-04 00-17-19](https://user-images.githubusercontent.com/3160384/222820556-b19b7efc-09ee-4efc-bc7d-8f2ec0767d92.png)

----

If we do the same in MySQL we will see `address`-related values in response:
```
echo 'create database companies;' > /tmp/init.sql
docker run --rm -v /tmp/init.sql:/docker-entrypoint-initdb.d/init.sql -p 13306:3306 -t -e MYSQL_ROOT_PASSWORD=SupErOassword mysql:8.0.32-debian
```

.. the same SQL ..

Result - `address`-related values  are present
![Screenshot from 2023-03-04 00-16-19](https://user-images.githubusercontent.com/3160384/222820588-8236e1e4-af90-43dd-b8fd-eaaa816be235.png)

**Comment by timsehn** [comment]
Great find. 

@max-hoffman is this one an easy fix?

**Comment by max-hoffman** [comment]
Here's the fix: https://github.com/dolthub/go-mysql-server/pull/1631

**Comment by nkonev** [comment]
Fixed in 0.75.0, thank you, team!
